draco/analysis/beamform.py

```diff
@@ -495,10 +497,20 @@ def _process_data(self, data):
         if "ra" in data.index_map:
             self.is_sstream = True
             self.ra = data.index_map["ra"]
+
+            # Calculate the epoch for the data so we can calculate the correct
+            # CIRS coordinates
+            if "lsd" not in data.attrs:
+                raise ValueError(
+                    "SiderealStream must have an LSD attribute to calculate the epoch."
+                )
+            self.epoch = self.telescope.lsd_to_unix(int(data.attrs["lsd"]))
```

Comment:
I think we need a provision here for when `data.attrs["lsd"]` is a list (sidereal stacks).
Alternatively, it could go in `icrs_to_cirs`.

Reply:
Thanks Mateus. That's an excellent point.